node-api/backend/genesis.go

```diff
@@ -26,12 +26,67 @@ import (
 	"github.com/berachain/beacon-kit/primitives/math"
 )
 
-// GetGenesis returns the genesis state of the beacon chain.
-func (b Backend) GenesisValidatorsRoot(slot math.Slot) (common.Root, error) {
-	// needs genesis_time and genesis_fork_version
-	st, _, err := b.stateFromSlot(slot)
+// GetGenesisValidatorsRoot returns the genesis validators root of the beacon chain.
+func (b *Backend) GenesisValidatorsRoot() (common.Root, error) {
+	// Fast path: read lock for checking cached value
```

Comment:
use atomic.Pointer[common.Root] to clean up, but looks correct (thread-safe wise)

Reply:
done!